### PR TITLE
Modernize compute and disk configuration

### PR DESCRIPTION
**Pull request by SaxonF**
## Summary

This is a precursor to unifying "compute and disk" and "infrastructure pages". First step is just updating the existing compute and disk page to make use of standard page and form patterns.

- Reorganizes the existing Compute and Disk form into the established settings layout pattern, with Scaling, Compute, Disk, and Advanced sections.
- Moves billing deltas to section headers while retaining the sticky review footer.
- Adds animated notices, validation-error scrolling, and a responsive compute-size selector (2 columns by default, 3 from 680px, and 4 from 900px).
- Preserves permissions, entitlements, cooldowns, read-only warnings, replica pricing, GP3 validation, Nano/PITR locks, and the free Micro upgrade treatment.
- Keeps the redesigned experience on `/settings/compute-and-disk` so it can be reviewed independently of the route cutover.

## Stack

1. #48368 (this PR)
2. #48369
3. #48370

## How to test

1. Check out `chore/infra-compute-1-config` and start Studio with `pnpm dev:studio`.
2. Open `/project/<ref>/settings/compute-and-disk`.
3. Confirm the page header and Scaling, Compute, Disk, and Advanced sections follow the standard settings-page spacing, with the plan notice directly below the Scaling header.
4. Change the compute size and disk configuration. Confirm billing deltas appear in the relevant section headers and the sticky review footer summarizes and applies the pending changes.
5. Enter invalid GP3 IOPS or throughput values and submit. Confirm the validation message is shown and the first invalid field is scrolled into view.
6. Exercise representative project states: insufficient permissions, read-only mode, resize cooldown, Nano with PITR, a project with read replicas, and a free Micro upgrade. Confirm the existing locks, warnings, pricing, and upgrade treatment remain intact.
7. Resize the viewport and confirm the compute cards use 2 columns by default, 3 columns from 680px, and no more than 4 columns from 900px.

## Summary by CodeRabbit

- **New Features**
  - Reworked the Compute and Disk settings UI into clearer sectioned pages, including smoother navigation to the first validation error.
  - Added a transitional in-form NoticeBar for consistent alert rendering.
  - Improved billing change badges to reflect accurate before/after totals.
- **Bug Fixes**
  - Fixed disk IOPS/throughput pricing to include replica-based charges.
  - Strengthened disk sizing validation (GP3 limits, legacy/size edge cases, spend-cap behavior, and provider-specific constraints).
- **Refactor**
  - Updated key form field layouts for improved readability.
- **Tests**
  - Expanded schema validation and pricing/billing badge test coverage.

### CODE REVIEW

File: apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts [hunk 1]
```diff
@@ -26,10 +26,26 @@ const COMPUTE_SIZES_BELOW_LARGE: Array<ComputeInstanceAddonVariantId> = [
...
+  newPrice: string | number
+}) {
+  return isDirty && !hasComputeSizeError && Number(oldPrice) !== Number(newPrice)
+}
+
 export function useDiskManagementReviewChanges(
   form: UseFormReturn<DiskStorageSchemaType>,
   numReplicas: number
 ) {
+  form.watch(['computeSize', 'totalSize', 'storageType', 'provisionedIOPS', 'throughput'])
```
Comment (alaister) [inline apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts]: `watch` re-renders the component that called `useForm`, not the caller. That's fine at the `DiskManagementForm.tsx:184` call site, but the other consumer — `DiskManagementReviewAndSubmitDialog.tsx:90` — receives `form` as a prop, so it's leaning on the parent re-rendering it. Works today, goes stale the moment either side gets memoized. `useWatch` subscribes the calling component directly.

Also worth consuming the returned values instead of the `getValues()` calls below: right now the watch list and the reads are hand-synced and already drifting — `growthPercent`, `minIncrementGb` and `maxSizeGb` are read at L114–127 but aren't in the watch list.

```suggestion
  useWatch({
    control: form.control,
    name: ['computeSize', 'totalSize', 'storageType', 'provisionedIOPS', 'throughput'],
  })
```

Needs the import on L2 updated to `import { useFormState, useWatch, type UseFormReturn } from 'react-hook-form'` (outside the diff, so I can't suggest it inline).

File: apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts [hunk 2]
```diff
@@ -86,6 +102,29 @@ export function useDiskManagementReviewChanges(
     Number(iopsPrice.newPrice) +
     Number(throughputPrice.newPrice)
 
+  const advancedBeforePrice = Number(iopsPrice.oldPrice) + Number(throughputPrice.oldPrice)
+  const advancedAfterPrice = Number(iopsPrice.newPrice) + Number(throughputPrice.newPrice)
+
+  const { isDirty, errors } = form.formState
```
Comment (alaister) [inline apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts]: Same ownership caveat as `watch` — `formState` is a Proxy that subscribes the `useForm` owner, not this hook's caller. Since these now gate the new badges, worth subscribing locally.

```suggestion
  const { isDirty, errors } = useFormState({ control: form.control })
```

File: apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts (hunk 1 above)
Comment (SaxonF) [reply apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog/DiskManagementReviewAndSubmitDialog.hooks.ts]: Thanks! Updated

**Review by alaister — APPROVED** [review]
Tested this on the staging preview and locally – everything in "How to test" checks out.

| # | Test | Result |
| --- | --- | --- |
| 3 | Page header + Scaling / Compute / Disk / Advanced sections, plan notice under Scaling | ✅ Sections render in order, notice sits directly under the Scaling heading |
| 4 | Billing deltas in section headers + sticky review footer | ✅ `$9.68 › $14.83/month` in the Compute header and footer, correct on the first render. Dialog totals line up (`+$99.54`), Cancel reverts cleanly |
| 5 | Invalid values + submit → validation message, scroll to first invalid field | ✅ Scroll works on submit – `rect.top` went 1729 → 320. See note below on GP3 |
| 6 | Representative project states | ⚠️ Partial – free Micro upgrade treatment confirmed, rest weren't reproducible |
| 7 | Responsive compute cards (2 / 3 / 4 cols) | ✅ Confirmed at 4 container widths. Worth noting these are container queries, so they key off the grid's own width, not the viewport |

Also checked the loading and error states of the compute grid by stubbing the addons request – skeletons render 6-up in the grid, and the error admonition renders full width. Neighbouring settings pages and a nav round-trip are all clean, no console errors.

**A couple I couldn't get to:** GP3 IOPS/throughput are gated behind `isSpendCapEnabled` in `disableDiskInputs`, which disables them regardless of compute size – so I couldn't exercise the GP3 validation path specifically. I verified the scroll-to-error mechanism itself using Max disk size instead. Read replicas, resize cooldown, read-only and Nano+PITR weren't reproducible on a test account either.

**I pushed a commit** (15f96116). It builds on the `useWatch`/`useFormState` change you already made in 576e5313:

- `useWatch` and `useFormState` now take a `name` list and the returned values get consumed, instead of re-reading through `getValues()` / `form.formState.defaultValues`. The `defaultValues` reads were still going through the `useForm`-owned Proxy, so they weren't actually subscribed in the dialog – and keeping the watch list and the read list in sync by hand had already drifted (`growthPercent`, `minIncrementGb` and `maxSizeGb` were read but never watched)
- Guarded the `NaN` that `valueAsNumber` produces on an empty disk size input – it was landing in form state and flowing into the price calcs and replica sizing
- Flattened the nested `isLoading ? … : addonsError ? … : (…)` in `ComputeSizeField` into named guards, since it wrapped ~200 lines

**Few things I noticed that predate this PR** – all confirmed by stashing my changes and re-testing, so none of them are from this branch. Not blockers, but the first one might be worth a look since this PR makes it a lot more visible:

- `calculateComputeSizePrice` only applies the nano → micro override to `oldComputeSize`, never to `newComputeSize` (`DiskManagement.utils.ts:64-73`). On a paid-plan project running nano that means any dirty form shows a phantom `Compute size $9.68 › $0.00/month` even when compute wasn't touched, and the footer total drops the base compute cost. Master's footer just said "N changes" so it didn't show – with a price badge it's pretty noticeable. Worth folding into this PR or a follow-up, wdyt?
- The three autoscale fields come back as `null` from the API but display `50` / `4` / `60000` as placeholders, so `form.reset()` on footer Cancel flips them controlled → uncontrolled and the typed value sticks in the DOM. Also means the Growth percent row can't be undone by hand – you get a `50% → 50%` row that won't go away. Fires a React warning each time
- The Advanced autoscale fields aren't disabled on a free plan, so they're editable under the "Only available on Pro Plan and above" notice. Long-standing, but this PR drops the collapsible so they're visible without clicking now
